src/auto/dbus_interface.rs

```diff
@@ -0,0 +1,71 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use gio_sys;
+use glib::object::IsA;
+use glib::translate::*;
+use std::fmt;
+use DBusInterfaceInfo;
+use DBusObject;
+
+glib_wrapper! {
+    pub struct DBusInterface(Interface<gio_sys::GDBusInterface>);
+
+    match fn {
+        get_type => || gio_sys::g_dbus_interface_get_type(),
+    }
+}
+
+pub const NONE_DBUS_INTERFACE: Option<&DBusInterface> = None;
+
+pub trait DBusInterfaceExt: 'static {
+    fn dup_object(&self) -> Option<DBusObject>;
+
+    fn get_info(&self) -> Option<DBusInterfaceInfo>;
+
+    fn get_object(&self) -> Option<DBusObject>;
+
+    fn set_object<P: IsA<DBusObject>>(&self, object: Option<&P>);
+}
+
+impl<O: IsA<DBusInterface>> DBusInterfaceExt for O {
+    fn dup_object(&self) -> Option<DBusObject> {
```

Comment:
`dup_object()` should be renamed to `get_object()`, and `get_object()` be hidden. `get_object()` is not safe, see the docs, and otherwise equivalent.

Reply:
tbh I'm not sure about the "rename" part. Don't like it when bindings are that unfaithful to the original library :D

Reply:
From a bindings point of view both functions work exactly the same, and both are "dup"licating (i.e. `g_object_ref()`) the return value. Just that the `get()` variant is internally unsafe while the `dup()` one is not (the `get()` one was a API design bug from what I understand).

So we definitely don't want to expose the `get()` variant, and as `dup()` makes semantically no sense for the bindings renaming it to `get()` would seem better and less confusing for users :)